packages/theme/src/provider.tsx

```diff
@@ -1,23 +1,60 @@
-import { FC } from 'react'
-import {
+import React, { FC, PropsWithChildren, useEffect } from 'react'
+import styled, {
   ThemeProvider as StyledThemeProvider,
   ThemeProviderProps,
 } from 'styled-components'
-import { themeDefault, themeDark, themeLight } from './themes'
+import { themeDark, themeDefault, themeLight } from './themes'
 import { Theme } from './types'
+import { initColors } from './document-head-contents'
+import { generateCssColorVariables } from './document-head-contents/utils/generate-css-color-variables'
+import { useThemeToggle } from '@lidofinance/cookie-theme-toggler'
 
-export const ThemeProvider: FC<ThemeProviderProps<Theme>> = (props) => {
-  const { theme = themeDefault, ...rest } = props
+// you need this if you're using themes, so we initiate this automatically
+initColors()
 
-  return <StyledThemeProvider theme={theme} {...rest} />
+const StyledWrapper = styled.div<{ colors?: Record<string, string> }>`
+  display: contents;
+  ${({ colors }) => (colors ? generateCssColorVariables(colors) : null)}
+`
+export const ThemeProvider: FC<
+  PropsWithChildren<{ theme: Theme; isAutoDetectedTheme?: boolean }>
+> = ({
+  theme = themeDefault,
+  children,
+  isAutoDetectedTheme = false,
+  ...rest
+}) => {
+  const parentTheme = useThemeToggle()
+  useEffect(() => {
+    if (typeof window === 'undefined') {
```

Comment:
We don't need to check for window in useEffects 

https://nextjs.org/docs/basic-features/data-fetching/client-side#client-side-data-fetching-with-useeffect

Reply:
you are right but thinking wrong way - beware that in that manner you are vendor-locking us to next.js